tests/datalab/test_datalab.py

```diff
@@ -980,53 +980,109 @@ def test_pred_probs_precedence(self, pred_probs, random_embeddings):
+
+        assert "label" in summary["issue_type"].values
+        assert (summary[summary["issue_type"] == "label"]["num_issues"] == 40).all()
+        assert np.isclose(
+            summary[summary["issue_type"] == "label"]["score"].values[0], 0.672974, atol=1e-5
+        )
 
+    def test_regression_with_predictions(self, lab, regression_data):
+        """Test that the regression issue checks find 12 label issues, based on the
+        predictions of a model.
 
-def test_regression():
-    data = make_data()
-    X, y = data["X"], data["y"]
-    test_df = pd.DataFrame(X, columns=["c1", "c2", "c3"])
-    test_df["y"] = y
-    lab = Datalab(data=test_df, label_name="y", task="regression")
+        Instead of running a model, we use the ground-truth to emulate a perfect model's predictions.
+
+        Testing the default behavior, we expect to find some label issues with a given mean score.
+        Increasing a threshold for flagging issues will flag more issues, but won't change the score.
+        """
+
+        # Use ground-truth to emulate a perfect model's predictions
+        y_pred = regression_data["true_y"]
+
+        lab.find_issues(predictions=y_pred)
+        summary = lab.get_issue_summary()
+        assert (summary[summary["issue_type"] == "label"]["num_issues"] == 12).all()
+        assert np.isclose(
+            summary[summary["issue_type"] == "label"]["score"].values[0], 0.58768, atol=1e-5
+        )
 
-    assert set(lab.list_default_issue_types()) == set(["label"])
-    assert set(lab.list_possible_issue_types()) == set(["label"])
+        # Apply a threshold for flagging issues. A larger threshold will flag more issues,
+        # but won't change the score.
+        lab.find_issues(predictions=y_pred, issue_types={"label": {"threshold": 0.5}})
+        summary = lab.get_issue_summary()
+        assert (summary[summary["issue_type"] == "label"]["num_issues"] == 22).all()
+        assert np.isclose(
+            summary[summary["issue_type"] == "label"]["score"].values[0], 0.58768, atol=1e-5
+        )
 
-    lab.find_issues(features=X)
-    lab.report()
-    summary = lab.get_issue_summary()
+    def test_regression_with_model_and_features(self, lab, regression_data):
+        """Test that the regression issue checks find label issue with another model."""
+        from sklearn.neighbors import KNeighborsRegressor
 
-    assert "label" in summary["issue_type"].values
-    assert (summary[summary["issue_type"] == "label"]["num_issues"] == 40).all()
+        model = KNeighborsRegressor(n_neighbors=5)
+        X = regression_data["X"]
+        lab.find_issues(
+            features=X, issue_types={"label": {"clean_learning_kwargs": {"model": model}}}
+        )
+        summary = lab.get_issue_summary()
+        assert (summary[summary["issue_type"] == "label"]["num_issues"] == 30).all()
+        assert np.isclose(
+            summary[summary["issue_type"] == "label"]["score"].values[0], 0.777305, atol=1e-5
+        )
```

Comment:
can we assert which of the datapoints have issues, best test is assert that checks for high jaccard similarity between set of flagged issues and set of true issues

Reply:
this is not an end-to-end test though of what matters. What matters is that this code can accurately detect label issues. I'm not sure where that's being tested (maybe I overlooked)

Reply:
I've updated the dataset for testing and asserted which datapoints have issues. So this should be resolved.

The tricky thing is that the "real issues" might have smaller residuals and are not picked up (leading to a lower Jaccard similarity).

Reply:
The tests checking the high Jaccard similarity were placed behind those lines.

But now the summary assertions have been removed.
I've added some Jaccard Similarity assertions, as well as low FPR assertions as well.